tests/components/sensor/test_sql.py

```diff
@@ -35,3 +36,22 @@ def test_query(self):
 
         state = self.hass.states.get('sensor.count_tables')
         self.assertEqual(state.state, '0')
+
+    def test_no_results_query(self):
```

Comment:
You should test the opposite. This case is already covered by all the other tests.

Reply:
This is the test I missed in the first place (which led to the issue)... don't understand you :(